azurerm/internal/services/desktopvirtualization/virtual_desktop_workspace_application_group_association_resource.go

```diff
@@ -0,0 +1,182 @@
+	}
+	output = append(output, applicationGroupReferenceID)
+
+	read.ApplicationGroupReferences = &output
+
+	if _, err = client.CreateOrUpdate(ctx, resourceGroup, workspaceName, read); err != nil {
+		return fmt.Errorf("Updating Virtual Desktop Workspace Association for Application Group %q (Resource Group %q): %+v", workspaceName, resourceGroup, err)
+	}
+
+	d.SetId(resourceID)
+
+	return resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationRead(d, meta)
+}
+
+func resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.WorkspacesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	splitID := strings.Split(d.Id(), "|")
+	if len(splitID) != 2 {
+		return fmt.Errorf("Expected ID to be in the format {workspaceID}/{networkSecurityGroup} but got %q", d.Id())
+	}
+
+	applicationGroupReferenceID := d.Get("application_group_reference_id").(string)
+
+	wsID, err := parse.DesktopVirtualizationWorkspaceID(splitID[0])
+	if err != nil {
+		return err
+	}
+
+	agID, err := parse.VirtualDesktopApplicationGroupID(splitID[1])
+	if err != nil {
+		return err
+	}
+
+	locks.ByName(wsID.Name, "Microsoft.DesktopVirtualization/workspaces")
+	defer locks.UnlockByName(wsID.Name, "Microsoft.DesktopVirtualization/workspaces")
+
+	locks.ByName(agID.Name, "Microsoft.DesktopVirtualization/applicationgroups")
+	defer locks.UnlockByName(agID.Name, "Microsoft.DesktopVirtualization/applicationgroups")
+
+	read, err := client.Get(ctx, wsID.ResourceGroup, wsID.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(read.Response) {
+			return fmt.Errorf("Virtual Desktop Workspace %q (Resource Group %q) was not found", wsID.Name, wsID.ResourceGroup)
+		}
+
+		return fmt.Errorf("Retrieving Virtual Desktop Workspace %q (Resource Group %q): %+v", wsID.Name, wsID.ResourceGroup, err)
+	}
+
+	refs := read.ApplicationGroupReferences
+	if refs == nil {
+		return fmt.Errorf("ApplicationGroupReferences was nil for Virtual Desktop Workspace %q (Resource Group %q)", wsID.Name, wsID.ResourceGroup)
+	}
+
+	output := make([]string, 0)
+	output = append(output, *refs...)
+	output = utils.RemoveFromStringArray(output, applicationGroupReferenceID)
+
+	read.ApplicationGroupReferences = &output
+
+	if _, err = client.CreateOrUpdate(ctx, wsID.ResourceGroup, wsID.Name, read); err != nil {
+		return fmt.Errorf("Updating Virtual Desktop Workspace Association for Application Group %q (Resource Group %q): %+v", wsID.Name, wsID.ResourceGroup, err)
+	}
+
+	return nil
+}
+
+func contains(s []string, r string) bool {
+	for _, a := range s {
+		if a == r {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
There is an existing function in the provider we can use instead: utils.StringSliceContains i beleive.

Reply:
I will take a look at this function. Thanks!

Reply:
I can't see this in the repo:
https://github.com/terraform-providers/terraform-provider-azurerm/search?q=StringSliceContains
We couldn’t find any code matching 'StringSliceContains' in terraform-providers/terraform-provider-azurerm

Reply:
Would you like me to add it to utils?